### PR TITLE
Implement DEDUP function

### LINKED ISSUE #1426: Implement `DEDUP` function
## Description
`DEDUP(list)`
Remove duplicate elements in list.

## Examples
```sql
SELECT DEDUP(CAST('[0, 1, 2, 2, 3, 4, 4]') AS List)
-- > [0, 1, 2, 3, 4]
```
```sql
SELECT DEDUP(CAST('["1", 1, 1, "1", "1"]') AS List)
-- > ["1", 1, "1"]
```

### PR DESCRIPTION
**Pull request by jinlee0**
resolves #1426

### CODE REVIEW

File: core/src/executor/evaluate/function.rs [hunk 1]
```diff
@@ -1065,3 +1066,14 @@ pub fn splice<'a>(
...
 }
+
+pub fn dedup<'a>(list: Evaluated<'_>) -> ControlFlow<Evaluated<'a>> {
+    let list = match list.try_into().break_if_null()? {
+        Value::List(list) => list,
+        _ => {
+            return Err(EvaluateError::ListTypeRequired.into()).into_control_flow();
+        }
+    };
+    let result = list.into_iter().dedup_by(|a, b| a == b).collect::<Vec<_>>();
```
[the remark this answers is not in the document]
Comment (panarch) [reply core/src/executor/evaluate/function.rs:1077]: everything else looks nice!

**Review by panarch — CHANGES_REQUESTED** [review]

**Review by zmrdltl — COMMENTED** [review]
resolve #1426

**Comment by jinlee0** [comment]
Sorry, I'm late

**Review by panarch — APPROVED** [review]
Looks all nice :D Thanks a lot 👍

**Pull request merged**